### PR TITLE
[ERSSUP-61789]-[]-[Prevent user-restricted from using AUTHORISED_APPL…

### PR DESCRIPTION
**Pull request by jamiegreen4**
…ICATION]-[JG]

## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: proxies/live/apiproxy/targets/ers-target.xml [hunk 1]
```diff
@@ -78,6 +56,64 @@
...
+                    <Name>AssignMessage.Swap.CorrelationHeader</Name>
+                </Step></Request>
+            <Response/>
+        </Flow>
+        <!-- App restricted not yet supported -->
+        <Flow name="app-restricted-flow">
+            <Condition>(accesstoken.auth_type == "app")</Condition>
+            <Request>
+                <Step>
+                    <Name>RaiseFault.400BadRequest</Name>
```
Comment (nhsd-jack-wainwright) [inline proxies/live/apiproxy/targets/ers-target.xml:102]: As mentioned previously I wonder if a 403 would make more sense than a 400 in this scenario?

File: tests/data.py [hunk 2]
```diff
@@ -7,6 +7,7 @@ class Actor(Enum):
     RCA = ("555031998104", "D82106", "REFERRING_CLINICIAN_ADMIN")
     SPC = ("555032006106", "RCD", "SERVICE_PROVIDER_CLINICIAN")
     SPCA = ("555031998104", "RCD", "SERVICE_PROVIDER_CLINICIAN_ADMIN", "555031999105")
+    AA = ("555031998104", "RCD", "AUTHORISED_APPLICATION")
```
Comment (nhsd-jack-wainwright) [inline tests/data.py:10]: I don't think this user would ever be valid currently, I wonder if its worth including a comment highlighting that this actor isn't usable for user-restricted access.

File: tests/sandbox/test_get_codesystem.py [hunk 3]
```diff
@@ -14,7 +14,27 @@
 class TestGetCodesystem(SandboxTest):
     @pytest.fixture
     def unauthorised_actors(self) -> Iterable[Actor]:
-        return []
+        return self.unauthorised_actors_list()
+
+    def unauthorised_actors_list(self) -> List[Actor]:
```
Comment (nhsd-jack-wainwright) [inline tests/sandbox/test_get_codesystem.py:19]: Could the result of this method be moved into the existing method above?

Comment (jamiegreen4) [reply tests/sandbox/test_get_codesystem.py:19]: pytest gave me an error for calling the fixture directly. One of their suggested workarounds was to have the fixture call a method

File: proxies/live/apiproxy/targets/ers-target.xml (hunk 1 above)
Comment (jamiegreen4) [reply proxies/live/apiproxy/targets/ers-target.xml:102]: I don't mind changing this but probably worth getting Adam's view on this

Comment (adamoldfield-nhs) [reply proxies/live/apiproxy/targets/ers-target.xml:102]: Yeah I agree 403.

Comment (jamiegreen4) [reply proxies/live/apiproxy/targets/ers-target.xml]: Changed

**Review by nhsd-jack-wainwright — APPROVED** [review]
Looks good to me 🙂

**Pull request merged**